File: backend/app/routers/doctors.py
from datetime import date

from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session

from ..database import get_db
from ..models import Doctor, VisitMode
from ..schemas import DoctorOut, SlotOut
from ..services.scheduling import available_slots

router = APIRouter(prefix="/api/doctors", tags=["doctors"])


@router.get("", response_model=list[DoctorOut])
def list_doctors(db: Session = Depends(get_db)):
    return db.query(Doctor).all()


@router.get("/{doctor_id}", response_model=DoctorOut)
def get_doctor(doctor_id: str, db: Session = Depends(get_db)):
    doc = db.query(Doctor).filter(Doctor.id == doctor_id).first()
    if not doc:
        raise HTTPException(status_code=404, detail="Doctor not found")
    return doc


@router.get("/{doctor_id}/slots", response_model=list[SlotOut])
def get_slots(
    doctor_id: str,
    day: date = Query(..., description="YYYY-MM-DD"),
    mode: VisitMode | None = None,
    db: Session = Depends(get_db),
):
    if not db.query(Doctor).filter(Doctor.id == doctor_id).first():
        raise HTTPException(status_code=404, detail="Doctor not found")
    return available_slots(db, doctor_id, day, mode)

File: server/routes/doctors.js
const { Router } = require("express");
const db = require("../db");
const { availableSlots } = require("../scheduling");

const router = Router();

router.get("/", (req, res) => {
  const docs = db.prepare("SELECT * FROM doctors").all();
  res.json(docs.map((d) => ({
    id: d.id,
    specialty: d.specialty,
    bio: d.bio,
    qualifications: d.qualifications,
    consult_fee: d.consult_fee,
    telehealth_enabled: !!d.telehealth_enabled,
  })));
});

router.get("/:id", (req, res) => {
  const doc = db.prepare("SELECT * FROM doctors WHERE id = ?").get(req.params.id);
  if (!doc) return res.status(404).json({ detail: "Doctor not found" });
  res.json({
    id: doc.id,
    specialty: doc.specialty,
    bio: doc.bio,
    qualifications: doc.qualifications,
    consult_fee: doc.consult_fee,
    telehealth_enabled: !!doc.telehealth_enabled,
  });
});

router.get("/:id/slots", (req, res) => {
  const doc = db.prepare("SELECT id FROM doctors WHERE id = ?").get(req.params.id);
  if (!doc) return res.status(404).json({ detail: "Doctor not found" });
  const { day, mode } = req.query;
  if (!day) return res.status(400).json({ detail: "day query parameter required (YYYY-MM-DD)" });
  const slots = availableSlots(doc.id, day, mode || null);
  res.json(slots);
});

module.exports = router;
